src/three-idx.ts

```diff
@@ -89,15 +92,18 @@ export class ThreeIDX {
   }
 
   async create3idDoc(docParams: ThreeIdState): Promise<void> {
-    this.docs.threeId = await TileDocument.create(
-      this.ceramic,
-      docParams.content,
-      docParams.metadata,
-      {
+    if (!docParams.content) {
+      this.docs.threeId = (await this.loader.deterministic(docParams.metadata, {
         anchor: false,
         publish: false,
-      }
-    )
+      })) as TileDocument<Record<string, any>>
+    } else {
+      this.docs.threeId = await this.loader.create(docParams.content, docParams.metadata, {
```

Comment:
Is the expected behavior that it could be looked-up deterministically? If so I don't think this would work, as it will create the stream with contents and therefore a different hash than the deterministic metadata, no?

Shouldn't the logic be something like this?
```ts
this.docs.threeId = await this.loader.deterministic(docParams.metadata)
if (docParams.content != null) {
  await this.docs.threeId.update(docParams.content)
}
```

Reply:
3IDs are generally not created deterministically. The only exception is if it's a 3IDv0, then we create an empty 3ID document to start (because the keys can be looked up from the legacy document.
The reason we are not doing what you suggest above is that we would have to wait for an anchor until those keys become valid, which means that you could not sign anything with your 3ID until then.